src/api/app/services/workflows/yaml_downloader.rb

```diff
@@ -15,46 +15,42 @@ def call
     private
 
     def download_yaml_file
-      url = download_url
-      return download_gitlab_yaml_file if url.nil? && @scm_payload[:scm] == 'gitlab'
-
-      Down.download(url, max_size: MAX_FILE_SIZE)
-    rescue Down::Error => e
-      raise Token::Errors::NonExistentWorkflowsFile, "#{@token.workflow_configuration_url} could not be downloaded.\n#{e.message}" if @token.workflow_configuration_url.present?
-
-      # 'target_branch' can contain a commit sha (when tag push) instead of a branch name
-      raise Token::Errors::NonExistentWorkflowsFile, "#{@token.workflow_configuration_path} could not be downloaded from the SCM branch/commit #{@scm_payload[:target_branch]}." \
-                                                     "\nIs the configuration file in the expected place? Check #{DOCUMENTATION_LINK}\n#{e.message}"
-    end
-
-    def download_url
       # When an external URL is given, it prevails over the path.
-      return @token.workflow_configuration_url if @token.workflow_configuration_url.present?
+      return download_from_url(@token.workflow_configuration_url) if @token.workflow_configuration_url.present?
 
       case @scm_payload[:scm]
-      when 'github'
-        client = Octokit::Client.new(access_token: @token.scm_token, api_endpoint: @scm_payload[:api_endpoint])
-        # :ref can be the name of the commit, branch or tag.
-        client.content("#{@scm_payload[:target_repository_full_name]}", path: "/#{@token.workflow_configuration_path}", ref: @scm_payload[:target_branch])[:download_url]
       when 'gitea'
-        gitea_download_url
+        download_from_url(gitea_configuration_file_url)
```

Comment:
I would do this inside  `gitea_configuration_file_url` and call that method `download_gitea_yaml_file` for consistency.

Reply:
Done.